.github/workflows/codeql-analysis.yml

```diff
@@ -24,16 +24,24 @@ jobs:
       with:
         fetch-depth: 0
 
-    - run: git checkout HEAD^2
-      if: ${{ github.event_name == 'pull_request' }}
+    - name: Cache Tools
+      uses: actions/cache@v2
+      with:
+        path: tools
+        key: ${{ runner.os }}-tools-${{ hashFiles('recipe.cake') }}
```

Comment:
Not for this PR in particular, but in general, would it make sense to also cache the NuGet cache folder, so that dependencies that are imported via `PackageReference` are also cached?

I imagine you'd have to include the `.csproj` files in the hash as well to purge the cache when they change too. Prob. `global.json` as well 🤔 

e.g.
```yaml
key: ${{ runner.os }}-${{ hashFiles('recipe.cake', 'global.json', 'source/**/*.csproj') }}
```

Reply:
I think you're right.
I opened up https://github.com/cake-contrib/Cake.Recipe/issues/815 for that - I'd like to document that before we start implementing it 😄

Reply:
Don't think we would need to update the hash files would we? We run dotnet/NuGet restore each build, so if there are missing dependencies they will be pulled in without needing to bust the cache. No?